src/rard/static/js/fetchAndAddEditorModal.js

```diff
@@ -0,0 +1,74 @@
+async function getModalTemplate() {
+  try {
+    var currentPage = window.location.pathname;
+    const response = await fetch(`${modalUrl}?sourcePage=${currentPage}`);
+    return response.text();
+  } catch (error) {
+    console.error("Error fetching modal template:", error);
+    return null;
+  }
+}
+
+function addSymbolPickerToModal() {
+  var symbolPicker = document.body.querySelector("#myForm");
+  var modal = document.getElementById("editContentModal");
+  modal.appendChild(symbolPicker);
+}
+
+function replaceSymbolPickerInBody() {
+  var originalSymbolPickerContainer =
+    document.body.querySelector(".container.mt-5");
+  var modalPicker = document
+    .getElementById("editContentModal")
+    .querySelector("#myForm");
+  originalSymbolPickerContainer.appendChild(modalPicker);
+}
+
+async function addModal(content) {
+  var modalContainer = document.createElement("div");
+  var modalHTML = await getModalTemplate();
+  modalContainer.innerHTML = modalHTML;
+  return new Promise((resolve) => {
+    document.body.appendChild(modalContainer);
+    $(modalContainer).find(".modal").modal({
+      backdrop: "static",
+    }); // init the modal
+    // init the editor
+    initRichTextEditor($("#modal-content-editor"));
+    setTimeout(() => {
```

Comment:
it makes sure that the modal doesn't try initialise immediately since the ql-editor needs to initialise first (as it needs to select for it). the way the editors are initialised means I couldn't assign it to a variable and say `if editor variable...`